core/chains/evm/logpoller/log_poller.go

```diff
@@ -482,6 +489,14 @@ func (lp *logPoller) LatestBlock(qopts ...pg.QOpt) (int64, error) {
 	return b.BlockNumber, nil
 }
 
+func (lp *logPoller) BlockByNumber(n int64, qopts ...pg.QOpt) (*LogPollerBlock, error) {
+	b, err := lp.orm.SelectBlockByNumber(n, qopts...)
+	if err != nil {
+		return nil, err
+	}
+	return b, nil
```

Comment:
Simplification: can just return `lp.orm.SelectBlockByNumber` directly, since caller will check `err != nil` prior to using `LogPollerBlock`.

Reply:
(hopefully 😛 )